src/test/kotlin/no/skatteetaten/aurora/databasehotel/testutils.kt

```diff
@@ -46,3 +46,45 @@ fun createOracleSchema(mangerDataSource: HikariDataSource): HikariDataSource {
     val schemaName = manager.createSchema(username, password)
     return DataSourceUtils.createDataSource(mangerDataSource.jdbcUrl, schemaName, password)
 }
+
+fun DatabaseManager.cleanPostgresTestSchemas(schemas: List<Schema>): List<Exception> {
+    val suppressedExceptions = arrayListOf<Exception>()
+    if (this is PostgresDatabaseManager) {
+        val roleRemoveFailed = arrayListOf<Schema>()
+
+        schemas.forEach {
+            try {
+                permanentlyDeleteSchema(it)
+            } catch (e: Exception) {
+                if (e.message!!.contains("cannot be dropped because some objects depend on it")) {
```

Comment:
Hva skjer her når permanentlyDeleteSchema feiler med en annen melding? Blir det bare ignorert? Og hva skjer neste gang testene kjører? er man da i en limbo state hvor noe er ryddet og annet ikke er og man må lete etter hvorfor det feiler. Jeg vil heller at man feiler når man ikke klarer å rydde opp, slik at man manuelt kan gå inn å korrigere evt.

Reply:
Dette er nå flyttet ut av testutils, jeg har implementert PostgresCleaner som bruker forskjellige strategier for opprydding. Exceptions blir ikke ignorert lenger, men kastes på slutten av oppryddingen.